core/domain/suggestion_services.py

```diff
@@ -593,12 +616,226 @@ def get_translation_suggestions_waiting_longest_for_review_per_lang(
+        Sample2 </p> will give as output: Sample1 [MATH] Sample2.
+    Note: similar logic exists in the frontend in format-rte-preview.filter.ts.
+
+    Args:
+        html_content_string: str. The content html string to convert to plain
+            text.
+
+    Returns:
+        str. The plain text string from the given html content string.
+    """
+
+    def _replace_rte_tag(rte_tag):
+        """Replaces all of the <oppia-noninteractive-**> tags.
+
+        Args:
+            rte_tag: MatchObject. A matched object that contins the
+                oppia-noninteractive rte tag.
+
+        Returns:
+            str. The string to replace the rte tag with.
+        """
+        # Convert the MatchObject to a string.
+        rte_tag_string = rte_tag.group(0)
+        # Get the name of the noninteractive tag (ex. math).
+        replace_string = rte_tag_string.split('-')[2].split(' ')[0]
+        if replace_string[-1] == '>':
+            replace_string = replace_string[:-1]
+        replace_string = replace_string.capitalize()
+        replace_string = ' [%s] ' % replace_string
+        return replace_string
+
+    # Replace all the opening <oppia-noninteractive-**> tags with their names
+    # capitalized in square brackets.
+    html_content_string_with_noninteractive_tag_replaced = re.sub(
+        r'<(oppia-noninteractive\s*?)[^>]+>', _replace_rte_tag,
+        html_content_string)
+    # Get rid of all of the other html tags, including closing tags and
+    # malformed html tags if they exist.
+    plain_text = html_cleaner.strip_html_tags(
+        html_content_string_with_noninteractive_tag_replaced
+    )
+    # Remove trailing and leading whitespace and ensure that all words are
+    # separated by a single space.
+    plain_text_without_contiguous_whitespace = ' '.join(plain_text.split())
+    return plain_text_without_contiguous_whitespace
+
+
+def create_reviewable_suggestion_email_info_from_suggestion(suggestion):
+    """Creates an object with the key information needed to notify reviewers or
+    admins that the given suggestion needs review.
+
+    Args:
+        suggestion: Suggestion. The suggestion used to create the
+            ReviewableSuggestionEmailInfo object. Note that the suggestion's
+            status must be in review.
+
+    Returns:
+        ReviewableSuggestionEmailInfo. The corresponding reviewable suggestion
+        email info.
+
+    Raises:
+        Exception. The keys in SUGGESTION_EMPHASIZED_TEXT_GETTER_FUNCTIONS
+            should be consist with the suggestion types offered on the
+            Contributor Dashboard.
+        Exception. The suggestion type must be offered on the Contributor
+            Dashboard.
+    """
+    # This check makes sure that an error is raised if a new suggestion type
+    # is added to the Contributor Dashboard but hasn't been added to
+    # SUGGESTION_EMPHASIZED_TEXT_GETTER_FUNCTIONS.
+    if set(suggestion_models.CONTRIBUTOR_DASHBOARD_SUGGESTION_TYPES) != set(
```

Comment:
This only needs to be checked in a backend test, not while the code is being run.

Reply:
Changed!

Reply:
Looks resolved.